src/lib/modes.js

```diff
@@ -73,12 +73,25 @@ export const COMMENT = function(begin, end, modeOptions = {}) {
     },
     modeOptions
   );
-  mode.contains.push(PHRASAL_WORDS_MODE);
+  // mode.contains.push(PHRASAL_WORDS_MODE);
   mode.contains.push({
     className: 'doctag',
-    begin: '(?:TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):',
+    // hack to avoid the space from being included. the space is necessary to
+    // match here to prevent the plain text rule below from gobbling up doctags
+    begin: '[ ]*(?=(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):)',
+    end: /(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):/,
+    excludeBegin: true,
     relevance: 0
   });
+  // looking like plain text, more likely to be a comment
+  mode.contains.push(
+    {
+      // TODO: how to include ", (, ) without breaking grammars that use these for
+      // comment delimiters?
+      // begin: /[ ]+([()"]?([A-Za-z'-]{3,}|is|a|I|so|us|[tT][oO]|at|if|in|it|on)[.]?[()":]?([.][ ]|[ ]|\))){3}/
+      begin: /[ ]+(([A-Za-z'-]{3,}|is|a|I|so|us|[tT][oO]|at|if|in|it|on)[.]?[:]?([.][ ]|[ ]|\))){3}/
```

Comment:
```suggestion
      // TODO: how to include ", (, ) without breaking grammars that use these for
      // comment delimiters?
      // begin: /[ ]+([()"]?([A-Za-z'-]{3,}|is|a|I|so|us|[tT][oO]|at|if|in|it|on)[.]?[()":]?([.][ ]|[ ]|\))){3}/
      //
      // see https://github.com/highlightjs/highlight.js/issues/2827
      begin: /[ ]+(([A-Za-z'-]{3,}|is|a|I|so|us|[tT][oO]|at|if|in|it|on)[.]?[:]?([.][ ]|[ ]|\))){3}/
```

I would add a link to the issue explaining what this regex works on. OR explaining how you reached this seemingly arbitrary list of words. The regex makes sense in the context of that URL and with the regex101 screenshot but not just from reading code.

Reply:
I will explain this further with comments then and we can link to the issue that's a good idea.  Basically it's just common 1 and 2 letters words since just matching `\w{1,2}` would be FAR too broad... I'm sure it's also missing a few words that we'll add over time (or not, since this is mostly an invisible helper for auto-detect).